shell/server/proxy.js

```diff
@@ -781,6 +783,23 @@ Proxy.prototype._callNewSession = function (request, viewInfo) {
         : [ "en-US", "en" ]
   });
 
+  return this.uiView.newSession(userInfo, makeHackSessionContext(this.grainId),
+    WebSession.typeId, params).session;
+};
+
+Proxy.prototype._callNewApiSession = function (request, userInfo) {
+  var self = this;
+
+  // TODO(someday): We are currently falling back to WebSession if we get any kind of error upon
+  // calling newSession with an ApiSession._id.
+  // Eventually we'll remove this logic once we're sure apps have updated.
+  return this.uiView.newSession(userInfo, makeHackSessionContext(this.grainId),
+    ApiSession.typeId).then(function (session) {
```

Comment:
Oh, I guess by the time I got done typing that, your change became ready to merge...